Mathlib/Order/CompleteLatticeIntervals.lean

```diff
@@ -220,4 +220,24 @@ noncomputable def Set.Icc.completeLinearOrder [ConditionallyCompleteLinearOrder
     {a b : α} (h : a ≤ b) : CompleteLinearOrder (Set.Icc a b) :=
   { Set.Icc.completeLattice h, Subtype.linearOrder _ with }
 
+lemma Set.Icc.coe_sSup [ConditionallyCompleteLattice α] {a b : α} (h : a ≤ b)
+    {S : Set (Set.Icc a b)} (hS : S.Nonempty) : letI := Set.Icc.completeLattice h
+    ↑(sSup S) = sSup ((↑) '' S : Set α) :=
+  congrArg Subtype.val (dif_neg hS.ne_empty)
+
+lemma Set.Icc.coe_sInf [ConditionallyCompleteLattice α] {a b : α} (h : a ≤ b)
+    {S : Set (Set.Icc a b)} (hS : S.Nonempty) : letI := Set.Icc.completeLattice h
+    ↑(sInf S) = sInf ((↑) '' S : Set α) :=
+  congrArg Subtype.val (dif_neg hS.ne_empty)
+
+lemma Set.Icc.coe_iSup [ConditionallyCompleteLattice α] {a b : α} (h : a ≤ b)
+    {β : Type*} [Nonempty β] {S : β → Set.Icc a b} : letI := Set.Icc.completeLattice h
+    ↑(iSup S) = iSup ((↑) ∘ S : β → α) :=
```

Comment:
I think we'd normally write the RHS here with `\infi` notation, especially since `∘` is not reducible any more.